rust/arrow/src/ffi.rs

```diff
@@ -275,19 +336,19 @@ fn bit_width(data_type: &DataType, i: usize) -> Result<usize> {
         }
         // Variable-sized binaries: have two buffers.
         // "small": first buffer is i32, second is in bytes
-        (DataType::Utf8, 1) | (DataType::Binary, 1) => size_of::<i32>() * 8,
-        (DataType::Utf8, 2) | (DataType::Binary, 2) => size_of::<u8>() * 8,
-        (DataType::Utf8, _) | (DataType::Binary, _) => {
+        (DataType::Utf8, 1) | (DataType::Binary, 1) | (DataType::List(_), 1) => size_of::<i32>() * 8,
+        (DataType::Utf8, 2) | (DataType::Binary, 2) | (DataType::List(_), 2) => size_of::<u8>() * 8,
```

Comment:
Lists don't have a buffer number 2.

Reply:
Hmm.. I thought it was a [validity bitmap and an offset buffer](https://arrow.apache.org/docs/format/Columnar.html?highlight=list%20buffer#variable-size-list-layout) and that the child data was counted differently.

What should be the correct number?

Reply:
The validity bitmap is buffer 0 and the offsets are buffer 1. You are defining a buffer 2 (of `u8`) which doesn't exist.

Reply:
The validity is the buffer 0, the offsets the buffer 1. The List has no buffer number two. If someone requests buffer 2 from a List Array, we should error instead.

Reply:
Right, Thanks!